validator/src/validator.rs

```diff
@@ -509,41 +513,25 @@ where
                 MappedReturn::Decision(block) => {
                     trace!("Tendermint returned block {}", block);
 
-                    // If the event is a result meaning the next macro block was produced we push it onto our local chain
-                    let block_copy = block.clone();
-
                     // Use a trusted push since these blocks were generated by this validator
+                    let block_publisher = BlockPublisher {
+                        network: Arc::clone(&self.network),
+                    };
                     let result = if cfg!(feature = "trusted_push") {
                         Blockchain::trusted_push(
                             self.blockchain.upgradable_read(),
                             Block::Macro(block),
+                            &block_publisher,
                         )
-                        .map_err(|e| error!("Failed to push macro block onto the chain: {:?}", e))
-                        .ok()
                     } else {
-                        Blockchain::push(self.blockchain.upgradable_read(), Block::Macro(block))
-                            .map_err(|e| {
-                                error!("Failed to push macro block onto the chain: {:?}", e)
-                            })
-                            .ok()
+                        Blockchain::push_with_hook(
+                            self.blockchain.upgradable_read(),
+                            Block::Macro(block),
+                            &block_publisher,
+                        )
                     };
-
-                    if result == Some(PushResult::Extended)
-                        || result == Some(PushResult::Rebranched)
-                    {
-                        if block_copy.is_election() {
-                            info!(
-                                block_number = &block_copy.header.block_number,
```

Comment:
You are losing the info logs I believe. Is that desired?

Reply:
I think it's fine to get rid of those.